unable to find resource module.vm in flat project structure

Hi,

i have a multi module project with flat project structure. In this case the module.vm file is found but velocity has problems loading the file as a resource.

In the attachment is a sample project which demonstrates the problem.

Here is the Stacktrace:
`[ERROR] Failed to execute goal com.github.zaplatynski:fsm-packagetype:2.3.0:moduleXml (default-moduleXml) on project fsm: Unable to find resource 'fsm/src/main/fsm/module.vm' -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.github.zaplatynski:fsm-packagetype:2.3.0:moduleXml (default-moduleXml) on project fsm: Unable to find resource 'fsm/src/main/fsm/module.vm'
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
	at org.codehaus.classworlds.Launcher.main(Launcher.java:47)
Caused by: org.apache.maven.plugin.MojoExecutionException: Unable to find resource 'fsm/src/main/fsm/module.vm'
	at com.github.zaplatynski.firstspirit.modules.fsm.velocity.ModuleXmlParser.parseModuleVm(ModuleXmlParser.java:68)
	at com.github.zaplatynski.firstspirit.modules.fsm.ModuleXmlMojo.execute(ModuleXmlMojo.java:57)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 21 more
Caused by: org.apache.velocity.exception.ResourceNotFoundException: Unable to find resource 'fsm/src/main/fsm/module.vm'
	at org.apache.velocity.runtime.resource.ResourceManagerImpl.loadResource(ResourceManagerImpl.java:474)
	at org.apache.velocity.runtime.resource.ResourceManagerImpl.getResource(ResourceManagerImpl.java:352)
	at org.apache.velocity.runtime.RuntimeInstance.getTemplate(RuntimeInstance.java:1533)
	at org.apache.velocity.runtime.RuntimeInstance.getTemplate(RuntimeInstance.java:1514)
	at org.apache.velocity.app.VelocityEngine.getTemplate(VelocityEngine.java:373)
	at com.github.zaplatynski.firstspirit.modules.fsm.velocity.VelocityManager.renderModuleXml(VelocityManager.java:43)
	at com.github.zaplatynski.firstspirit.modules.fsm.velocity.ModuleXmlParser.parseModuleVm(ModuleXmlParser.java:66)`
[fsm.zip](https://github.com/zaplatynski/fsm-packagetype/files/1124914/fsm.zip)

--- a/src/main/java/com/github/zaplatynski/firstspirit/modules/fsm/velocity/VelocityManager.java
+++ b/src/main/java/com/github/zaplatynski/firstspirit/modules/fsm/velocity/VelocityManager.java
@@ -86,7 +86,7 @@ public class VelocityManager {
 
   private static String getTemplateRoot(final MavenProject project) {
     if (project.getParent() != null && isNotFlatProject(project)) {
-        return project.getParent().getBasedir().getAbsolutePath();
+      return project.getParent().getBasedir().getAbsolutePath();
     }
     return project.getBasedir().getParentFile().getAbsolutePath();
   }

--- a/src/main/java/com/github/zaplatynski/firstspirit/modules/fsm/velocity/VelocityManager.java
+++ b/src/main/java/com/github/zaplatynski/firstspirit/modules/fsm/velocity/VelocityManager.java
@@ -8,6 +8,7 @@ import org.apache.velocity.runtime.RuntimeConstants;
 import org.apache.velocity.runtime.resource.loader.ClasspathResourceLoader;
 
 import java.io.Writer;
+import java.nio.file.Path;
 import java.util.Objects;
 import java.util.Properties;
 
@@ -84,9 +85,15 @@ public class VelocityManager {
   }
 
   private static String getTemplateRoot(final MavenProject project) {
-    if (project.getParent() != null) {
-      return project.getParent().getBasedir().getAbsolutePath();
+    if (project.getParent() != null && isNotFlatProject(project)) {
+        return project.getParent().getBasedir().getAbsolutePath();
     }
     return project.getBasedir().getParentFile().getAbsolutePath();
+  }
+
+  private static boolean isNotFlatProject(MavenProject project) {
+    Path basedir = project.getBasedir().toPath();
+    Path parentDir = project.getParent().getBasedir().toPath();
+    return basedir.startsWith(parentDir);
   }
 }

--- a/src/test/java/com/github/zaplatynski/firstspirit/modules/fsm/velocity/VelocityManagerTest.java
+++ b/src/test/java/com/github/zaplatynski/firstspirit/modules/fsm/velocity/VelocityManagerTest.java
@@ -61,6 +61,7 @@ public class VelocityManagerTest {
     templateRoot = temporary.getRoot();
     assertThat(templateRoot.exists(), is(true));
     when(parentProject.getBasedir()).thenReturn(templateRoot);
+    when(project.getBasedir()).thenReturn(templateRoot);
     when(project.getParent()).thenReturn(parentProject);
 
     final File parent = new File(templateRoot, "project/src/main/fsm/");
@@ -279,6 +280,7 @@ public class VelocityManagerTest {
     when(project.getProperties()).thenReturn(mavenProjectProperties);
 
     when(parentProject.getBasedir()).thenReturn(templateRoot);
+    when(project.getBasedir()).thenReturn(templateRoot);
     when(project.getParent()).thenReturn(parentProject);
 
     testling = new VelocityManager(project);

